mllib/src/main/scala/org/apache/spark/ml/clustering/GaussianMixture.scala

```diff
@@ -316,24 +319,129 @@ class GaussianMixture @Since("2.0.0") (
   @Since("2.0.0")
   def setSeed(value: Long): this.type = set(seed, value)
 
+  // number of samples per cluster to use when initializing Gaussians
+  private val nSamples = 5
+
   @Since("2.0.0")
   override def fit(dataset: Dataset[_]): GaussianMixtureModel = {
     transformSchema(dataset.schema, logging = true)
-    val rdd: RDD[OldVector] = dataset.select(col($(featuresCol))).rdd.map {
-      case Row(point: Vector) => OldVectors.fromML(point)
+
+    val sc = dataset.sparkSession.sparkContext
+    val _k = $(k)
```

Comment:
maybe `nClusters`? I don't have a strong preference but I think it's clearer when reading the code.